manifests/install.pp

```diff
@@ -19,4 +19,9 @@
     content => template('pulpcore/pulp-vhost.conf.erb'),
   } 
 
+  file { '/etc/pulp/settings.py':
```

Comment:
Usually this is considered configuration and goes in config.pp. The benefit is that you know the package is installed which usually means things like the config dir exist. In this case you probably need to ensure `/etc/pulp` is created as well.

Reply:
Now that there's automated tests, you can see this actually breaks. https://travis-ci.org/theforeman/puppet-pulpcore/jobs/614129962#L1000

```
  Error: Could not set 'file' on ensure: No such file or directory @ dir_s_mkdir - /etc/pulp/settings.py20191119-974-1mqifpv.lock (file: /etc/puppetlabs/code/modules/pulpcore/manifests/install.pp, line: 23)

  Error: Could not set 'file' on ensure: No such file or directory @ dir_s_mkdir - /etc/pulp/settings.py20191119-974-1mqifpv.lock (file: /etc/puppetlabs/code/modules/pulpcore/manifests/install.pp, line: 23)
```

The reason is that `/etc/pulp` doesn't exist. In `config.pp` I already ensure this is all set up using the correct paths.